src/xenia/base/memory.cc

```diff
@@ -34,15 +34,16 @@ void copy_and_swap_16_aligned(void* dest_ptr, const void* src_ptr,
                    0x04, 0x05, 0x02, 0x03, 0x00, 0x01);
 
   size_t i = 0;
-  size_t unaligned_words = (reinterpret_cast<uintptr_t>(src_ptr) & 0xF) / 2;
+  size_t unaligned_words =
+      (0x10 - (reinterpret_cast<uintptr_t>(src_ptr) & 0xF)) / sizeof(uint16_t);
   for (; unaligned_words > 0 && i < count; unaligned_words--, i++) {
```

Comment:
Nice catch! One thing I wanted to avoid was if `src_ptr` is actually aligned, this will come out with `unaligned_words = 16` and needlessly copy 16 words using the slow route.